### PR TITLE
Simplifies configuration of uktt resources

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

HOTT-<TODO>

### What?

I have added/removed/altered:

- [x] Pass all configuration to an instance of http client
- [x] Http client is passed to the resources rather than mixing configuration in a global
- [x] Simplify specs significantly and move them to their respective resource files

### Why?

I am doing this because:

- We had a race condition in the gem which will cause issues later down the line
- This gem was totally unworkable/difficult to see what methods have been tested and under what contexts
- Config was super confusing and made it painful to know what was being called

### AC

- The tests that are run are exactly the same - just with some significant tweaks

### CODE REVIEW

File: lib/uktt/export_chapter_pdf.rb [hunk 1]
```diff
@@ -1428,6 +1427,8 @@ def currency_symbol
 
   private
 
+  attr_reader :http
```
Comment (willfish) [inline lib/uktt/export_chapter_pdf.rb]: Remove me

File: lib/uktt.rb [hunk 2]
```diff
@@ -13,50 +13,18 @@
...
-  end
-
-  def self.config
-    @config
+  API_HOST_PROD           = 'https://www.trade-tariff.service.gov.uk'.freeze
+  API_HOST_LOCAL          = 'https://dev.trade-tariff.service.gov.uk'.freeze
+  API_VERSION             = 'v2'.freeze
+  DEFAULT_PARENT_CURRENCY = 'EUR'.freeze
+
+  def self.default_config
```
Comment (willfish) [inline lib/uktt.rb]: Remove me

File: lib/uktt/chapter.rb [hunk 3]
```diff
@@ -5,38 +5,24 @@ class Chapter < Base
...
     def retrieve_all
       fetch "#{RESOURCE_PATH}.json"
     end
 
-    def goods_nomenclatures
-      return '@chapter_id cannot be nil' if @chapter_id.nil?
-
-      fetch "goods_nomenclatures/chapter/#{@chapter_id}.json"
+    def goods_nomenclatures(chapter_id)
+      fetch "goods_nomenclatures/chapter/#{chapter_id}.json"
```
Comment (tavyy) [inline lib/uktt/chapter.rb:17]: Do we treat errors like chapter_id missing elsewhere?

File: lib/uktt/commodity.rb [hunk 4]
```diff
@@ -3,43 +3,20 @@ module Uktt
...
     end
 
     def find(id)
-      return '@response is nil, run #retrieve first' unless @response
-
-      response = @response.included.select do |obj|
-        obj.id === id || obj.type === id
-      end
-      response.length == 1 ? response.first : response
+      @response.included.find { |obj| obj.id == id }
```
Comment (tavyy) [inline lib/uktt/commodity.rb]: Should this not be `.select` here? They do seem to do 
```ruby
response.length == 1 ? response.first : response
```
which means they return the whole collection for cases where the array length is > 1 ?

File: lib/uktt/commodity.rb [hunk 5]
```diff
@@ -3,43 +3,20 @@ module Uktt
...
     end
 
     def find_in(arr)
-      return '@response is nil, run #retrieve first' unless @response
-
-      response = @response.included.select do |obj|
-        arr.include?(obj.id)
-      end
-      response.length == 1 ? response.first : response
+      @response.included.find { |obj| arr.include?(obj.id) }
```
Comment (tavyy) [inline lib/uktt/commodity.rb]: Same question as: https://github.com/trade-tariff/uktt/pull/6/files#r594173830

File: lib/uktt/commodity.rb (hunk 4 above)
Comment (tavyy) [reply lib/uktt/commodity.rb]: Also, the fact that no test has failed with this change makes me think they don't have proper unit tests for this case?

File: lib/uktt/commodity.rb (hunk 5 above)
Comment (willfish) [reply lib/uktt/commodity.rb]: Removed as not used

**Pull request merged**